harvester_e2e_tests/integration/test_images.py

```diff
@@ -0,0 +1,375 @@
+        """
+        wait_timeout = request.config.getoption('--wait-timeout')
+
+        create_image_url(api_client, qcow2_name, qcow2_image_url, wait_timeout)
+        create_image_url(api_client, iso_name, iso_image_url, wait_timeout)
+
+    @pytest.mark.images_p1
+    @pytest.mark.dependency(name="delete_image_recreate", depends=["create_image_url"])
+    def test_delete_image_recreate(self, api_client, request,
+                                   iso_image_url, unique_name, qcow2_name, fake_image_file):
+        """
+        Test create raw and iso type image from file
+
+        Steps:
+        1. Check the image created by URL exists
+        2. Delete the newly created image
+        3. Create an iso file type image from URL
+        4. Check the iso image exists
+        5. Upload an qcow2 file type image
+        5. Delete the newly uploaded file
+        6. Upload a new qcow2 file type image
+        """
+        wait_timeout = request.config.getoption('--wait-timeout')
+
+        get_image(api_client, qcow2_name)
+        delete_image(api_client, qcow2_name, wait_timeout)
+
+        create_image_url(api_client, qcow2_name, iso_image_url, wait_timeout)
+        get_image(api_client, qcow2_name)
+
+        resp = api_client.images.create_by_file(unique_name, fake_image_file)
+
+        assert 200 == resp.status_code, (
+            f"Failed to upload fake image with error:{resp.status_code}, {resp.content}"
+        )
+
+        get_image(api_client, unique_name)
+        delete_image(api_client, unique_name, wait_timeout)
+
+        resp = api_client.images.create_by_file(unique_name, fake_image_file)
+
+        assert 200 == resp.status_code, (
+            f"Failed to upload fake image with error:{resp.status_code}, {resp.content}"
+        )
+
+        get_image(api_client, unique_name)
+        delete_image(api_client, unique_name, wait_timeout)
+
+    @pytest.mark.images_p1
+    def test_create_invalid_file(self, api_client, unique_name, fake_invalid_image_file):
+        """
+        Test create upload image from invalid file type
+
+        Steps:
+        1. Try to upload invalid image file to images page, something like dmg, or tar.gzps
+        2. Check should get an error
+        """
+
+        resp = api_client.images.create_by_file(unique_name, fake_invalid_image_file)
+
+        assert 500 == resp.status_code, (
+            f"File size correct, it's a multiple of 512 bytes:{resp.status_code}, {resp.content}"
+        )
+
+    @pytest.mark.images_p1
+    @pytest.mark.dependency(name="edit_image_in_use", depends=["create_image_url"])
+    def test_edit_image_in_use(self, api_client, request, unique_name, qcow2_name):
+        """
+        Test can edit image which already in use
+
+        Steps:
+        1. Check the image created from URL exists
+        2. Create a VM and use the existing image created from url
+        3. Update the image labels and description
+        4. Check can change the image content
```

Comment:
To make the image `in use` state, we can just create a image volume from the image, it would be the same behavior when creating a VM with the image.